library/Garden/StaticCacheTrait.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * @author Alexander Kim <alexander.k@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL v2
+ */
+
+namespace Garden;
+
+/**
+ * For classes that need to cache some static values and configs.
+ *
+ */
+trait StaticCacheTrait {
+    /**
+     * @var array
+     */
+    protected static $sCache = [];
+
+    /**
+     * @var boolean
+     */
+    protected static $scInit = false;
+
+    /**
+     * Returns of cached value or calls self::f() function if key not found yet
+     *
+     * @param string $key Key to get
+     * @param mixed $default Default value for key to set if not set up
+     *
+     * @return mixed
+     */
+    public static function sc(string $key, $default = false) {
```

Comment:
The `c()` and `t()` functions in core are very special cases. Please don't establish a new precedent of abbreviation method names all over the place.